### ISSUE TITLE
DetailsList ARIA attributes are not allowed

### ISSUE THREAD

**Issue by sereymonyros** [microsoft/fluentui#16144]
Run accessibility insights app in DetailsList
Actual result: ARIA attributes are not allowed: aria-setsize="10", aria-posinset="1"
Expected result: should not see any accessibility issue at all
Repro: https://codepen.io/justslone-the-encoder/pen/vYXGJPM

Screenshot below:

![image](https://user-images.githubusercontent.com/8165490/101181484-c8671c80-3601-11eb-99f1-d81be3057be8.png)
![image](https://user-images.githubusercontent.com/8165490/101181516-d321b180-3601-11eb-8b7f-db88d6c3b141.png)

**Comment by JustSlone** [comment]
@sereymonyros thanks for filing this

Confirmed repro. Here's an isolated codepen: https://codepen.io/justslone-the-encoder/pen/vYXGJPM

So, this one is a bit complicated. 
In this case DetailsList is behaving more as a treegrid, and actually the absence of being able to use aria-setsize and aria-posinset was an omission from the ARIA 1.1 spec. [See this issue for more info](https://github.com/w3c/aria/issues/558)

Actually if you try out accessibility insights on the official treegrid examples it raises the same issues.
https://www.w3.org/TR/wai-aria-practices-1.1/examples/treegrid/treegrid-1.html
![image](https://user-images.githubusercontent.com/40304143/101268231-b1195380-3715-11eb-8529-30c647fc0b2b.png)


So, I'm tempted to say this is expected and OK, however, my one reservation is that DetailsList, even though it's behaving like a treegrid, is actually role="grid". 

@smhigley, You've had some thoughts or perspective on DetailsList's confused nature here, what are your thoughts on this issue?

**Comment by ms-kykwon** [comment]
@JustSlone Any luck on this issue?

**Comment by JustSlone** [comment]
@ms-kykwon, I believe this was fixed by another change. I'm not seeing repro anymore.

@smhigley might know more. 

I am seeing an error on the group selection checkbox, but that's a different issue.

**Comment by ecraig12345** [comment]
Group selection checkbox issue is tracked by #13385

**Comment by smhigley** [comment]
Both those issues should be resolved; I think this one can just be closed now, via #17300, and I'll go comment on the separate checkbox issue.

**Comment by ms-kykwon** [comment]
@JustSlone @ecraig12345 @smhigley GroupedList have a similar issue. Would you take a look at it, please? #17943
